File: leetcode2/1easy/최원준/Q999.java
package Leetcode;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q999 {
    class Solution {
        int[] dx = {0,0,-1,1};
        int[] dy = {1,-1,0,0};
        char[][] board;
        int n, m;

        public int isAttacking(int x, int y) {
            int ans = 0;
            for (int i=0; i<4; i++) {
                int nx = x+dx[i], ny = y+dy[i];
                while (nx >= 0 && ny >= 0 && nx < n && ny <m) {
                    if (board[nx][ny] == 'p') {
                        ans ++;
                        break;
                    } else if (board[nx][ny] != '.') break;
                    nx += dx[i];
                    ny += dy[i];
                }
            }
            return ans;
        }
        public int numRookCaptures(char[][] board) {
            this.board = board;
            this.n = board.length;
            this.m = board[0].length;

            for (int i=0; i<n; i++) for (int j=0; j<m; j++) {
                if (board[i][j] == 'R') {
                    return isAttacking(i,j);
                }
            }
            return -1;
        }
    }
}

File: leetcode2/1easy/강병호/Q999.py


<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    # 위 오 아 왼

    def numRookCaptures(self, board: list[list[str]]) -> int:

        dx = [0, 1, 0, -1]
        dy = [1, 0, -1, 0]
        result = 0
        def findRookLocIdx():
            for i in range(0, 8):
                for j in range(0, 8):
                    if board[i][j] == 'R':
                        return i, j
        
        loc_x, loc_y = findRookLocIdx()
        
        for i in range(0, 4):
            next_x, next_y = loc_x, loc_y
            while True:
                if next_x == -1 or next_x == 8 or next_y == -1 or next_y == 8:
                    break
                
                if board[next_x][next_y] == 'B':
                    break
                elif board[next_x][next_y] == 'p':
                    result+=1
                    break

                next_x += dx[i]
                next_y += dy[i]
        
        print(result)

        return result
    



if __name__ == "__main__":
<<<<<<< HEAD
    sol = Leetcode.이재훈.이재훈.Solution()
=======
    sol = Leetcode.이재훈.Solution()
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424

    sol.numRookCaptures([[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".","R",".",".",".","p"],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]])

    
